SofaKernel/modules/SofaBaseMechanics/MechanicalObject.inl

```diff
@@ -103,23 +104,18 @@ MechanicalObject<DataTypes>::MechanicalObject()
     , scale(initData(&scale, Vector3(1.0,1.0,1.0), "scale3d", "Scale of the DOFs in 3 dimensions"))
     , translation2(initData(&translation2, Vector3(), "translation2", "Translation of the DOFs, applied after the rest position has been computed"))
     , rotation2(initData(&rotation2, Vector3(), "rotation2", "Rotation of the DOFs, applied the after the rest position has been computed"))
-    , filename(initData(&filename, std::string(""), "filename", "File corresponding to the Mechanical Object", false))
-    , ignoreLoader(initData(&ignoreLoader, (bool) false, "ignoreLoader", "Is the Mechanical Object do not use a loader. (default=false)"))
+    , d_size(initData(&d_size, 0, "size", "Size of the vectors"))
+    , l_topology(initLink("topology","Link to the topology relevant for this object"))
     , f_reserve(initData(&f_reserve, 0, "reserve", "Size to reserve when creating vectors. (default=0)"))
-    , vsize(0)
     , m_gnuplotFileX(NULL)
     , m_gnuplotFileV(NULL)
 {
-    // HACK
-    if (!restScale.isSet())
```

Comment:
The Data is constructed here, there is now way it can be set at this point. So this is same as if(true)...
I have no idea why it was here in the first place. A guess would be that some code somewhere was not working if restScale was not flagged as set, and this hack ensures that this never happen...